jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -245,21 +156,25 @@ private String getToken() {
   /**
    * Authenticates permissions to pull.
    *
+   * @param credential the credential used to authenticate
    * @return an {@code Authorization} authenticating the pull
    * @throws RegistryAuthenticationFailedException if authentication fails
    */
-  public Authorization authenticatePull() throws RegistryAuthenticationFailedException {
-    return authenticate("pull");
+  public Authorization authenticatePull(@Nullable Credential credential)
```

Comment:
Sorry, one more question. Should we remove the `@Nullable` on `Credential`? Can we authenticate without credentials?

Reply:
Some registries require authorizing with an auth server even if the image is public, like public images on Docker Hub. And sometimes, it's inevitably null if no credential is configured, which may or may not lead to a failure. The auth server will decide if it will be authorized or not.

Reply:
For some reason I thought we had logic to decide if we needed to authorize at all?

Reply:
Yeah, there is some place something similar is happening. For example, first you just try pulling a public Docker Hub image, and if it fails (usually it does, if not always) and the registry returns with the auth challenge. Then we go to the auth server in the challenge to get the pull authorization. As a public image, you don't need any credentials. (Not setting the `Authorization` header) Passing null in this case will work as the image is public.